playground/ExternalServices/ExternalServices.AppHost/AppHost.cs

```diff
@@ -5,8 +5,19 @@
 
 var builder = DistributedApplication.CreateBuilder(args);
 
-// URL value is in appsettings.json, or can be overridden by environment variable
-var externalService = builder.AddExternalService("external-service", builder.AddParameter("external-service-url"));
+var externalServiceUrl = builder.AddParameter("external-service-url")
+    .WithDescription("The URL of the external service.")
+    .WithCustomInput(p => new()
```

Comment:
How do we feel about this extension method name? Seems pretty generic. In terms of the API design would it be better for the input to be passed to the callback pre-created like we do for endpoints?

Reply:
No more generic than WithDescription?

For the other point, do you mean making the Func accept the default input we build? To do that we'd need to add some more API to make it ergonomic to either mutate the input (some props are init only) or add a copy constructor and make the Func both accept and return the input. Not sure it's worth it but if others agree it's ok with me.

Reply:
Yes I like that callback takes the initial input

Reply:
@davidfowl lol can you be a bit more specific please 😉

Reply:
Of course not. Lets punt this for now, this PR is in great shape for 9.4, we can always add this later.